tests/EdgeToEdge/Routes/UpdateDonorRouteTest.php

```diff
@@ -4,88 +4,87 @@
 use WMDE\Fundraising\Frontend\Tests\Fixtures\FixedTokenGenerator;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\StoredDonations;
+use WMDE\Fundraising\Frontend\Tests\RebuildDatabaseSchemaTrait;
 
 /**
  * @covers \WMDE\Fundraising\Frontend\App\Controllers\Donation\UpdateDonorController
  */
 class UpdateDonorRouteTest extends WebRouteTestCase {
 
+	use RebuildDatabaseSchemaTrait;
+
 	private const PATH = 'donation/update';
 	private const CORRECT_UPDATE_TOKEN = 'b5b249c8beefb986faf8d186a3f16e86ef509ab2';
 	private const INVALID_UPDATE_TOKEN = '2ba905fe68e61f3a681d8faf689bfeeb8c942b5b';
 
 	public function testWhenCorrectPrivatePersonDataIsPosted_addressIsChanged(): void {
-		$this->markTestIncomplete( "This should work again when we finish updating the membership controllers" );
-
 		$this->modifyConfiguration( [ 'skin' => 'laika' ] );
-		$this->createEnvironment(
-			function ( Client $client, FunFunFactory $factory ): void {
-				$donation = $this->newStoredDonation( $factory );
-				$this->performRequest(
-					$client,
-					$this->newPrivateDonorData(),
-					$donation->getId(),
-					self::CORRECT_UPDATE_TOKEN,
-					self::CORRECT_UPDATE_TOKEN
-				);
-				$response = $client->getResponse();
-				$this->assertTrue( $response->isRedirect( $this->newValidSuccessRedirectUrl( $donation, $factory ) ) );
-
-				$crawler = $client->followRedirect();
-				$dataVars = $this->getDataApplicationVars( $crawler );
-				$this->assertEquals( $this->newPrivateDonorData()['addressType'], $dataVars->addressType );
-				$this->assertEquals( $this->newPrivateDonorData()['salutation'], $dataVars->address->salutation );
-				$this->assertEquals( $this->newPrivateDonorData()['firstName'], $dataVars->address->firstName );
-				$this->assertEquals( $this->newPrivateDonorData()['lastName'], $dataVars->address->lastName );
-				$this->assertEquals( $this->newPrivateDonorData()['street'], $dataVars->address->streetAddress );
-				$this->assertEquals( $this->newPrivateDonorData()['postcode'], $dataVars->address->postalCode );
-				$this->assertEquals( $this->newPrivateDonorData()['city'], $dataVars->address->city );
-				$this->assertEquals( $this->newPrivateDonorData()['country'], $dataVars->address->countryCode );
-				$this->assertEquals( $this->newPrivateDonorData()['email'], $dataVars->address->email );
-			}
+		$client = $this->createClient();
+		$factory = $this->getFactory();
+		$donation = $this->newStoredDonation( $factory );
+
+		$this->performRequest(
+			$client,
+			$this->newPrivateDonorData(),
+			$donation->getId(),
+			self::CORRECT_UPDATE_TOKEN,
+			self::CORRECT_UPDATE_TOKEN
 		);
+		$response = $client->getResponse();
+		$this->assertTrue( $response->isRedirect( $this->newValidSuccessRedirectUrl( $donation, $factory ) ) );
+
+		$crawler = $client->followRedirect();
+		$dataVars = $this->getDataApplicationVars( $crawler );
+		$this->assertEquals( $this->newPrivateDonorData()['addressType'], $dataVars->addressType );
```

Comment:
This could be optimised by assigning the array to a variable rather than recreating it over and over